File: Spring_Final_Project_04_01_26/Spring_Final_Project/src/main/java/com/My/Spring_Final_Project/Service/admin/AdminStatisticsService.java
package com.My.Spring_Final_Project.Service.admin;

import com.My.Spring_Final_Project.DTO.admin.AdminStatisticsDTO;
import com.My.Spring_Final_Project.DTO.admin.AdminUserStatisticsDTO;

public interface AdminStatisticsService {

    AdminUserStatisticsDTO getUserStatistics();

    AdminStatisticsDTO getSystemStatistics();
}

File: Angular_Final_project/src/app/admin/services/admin-statistics.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { AdminUserStatisticsDTO } from '../models/admin-user-statistics.model';
import { AdminStatisticsDTO } from '../models/admin-statistics.model';

@Injectable({
  providedIn: 'root'
})
export class AdminStatisticsService {
  private baseUrl = `${environment.apiUrl}/admin/statistics`;

  constructor(private http: HttpClient) {}

  getUserStatistics(): Observable<AdminUserStatisticsDTO> {
    return this.http.get<AdminUserStatisticsDTO>(`${this.baseUrl}/users`);
  }

  getSystemStatistics(): Observable<AdminStatisticsDTO> {
    return this.http.get<AdminStatisticsDTO>(`${this.baseUrl}/system`);
  }
}
